yamllint/rules/float_values.py

```diff
@@ -0,0 +1,162 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Use this rule to limit the permitted values for floats.
+YAML permits three classes of float expressions:
+
+`[-+]?(\\.[0-9]+|[0-9]+(\\.[0-9]*)?)([eE][-+]?[0-9]+)?`
+`[-+]?(\\.inf|\\.Inf|\\.INF)`
+`\\.nan|\\.NaN|\\.NAN`
+
+.. rubric:: Options
+
+* Use ``require-numeral-before-decimal'`` to require floats to start
+    with a numeral (ex 0.0 instead of .0)
+* Use ``forbid-scientific-notation`` to forbid scientific notation
+* Use ``forbid-not-a-number`` to forbid NaN values
+* Use ``forbid-infinity`` to forbid infinite values
+
+.. rubric:: Default values (when enabled)
+
+.. code-block:: yaml
+
+ rules:
+   float-values:
+        require-numeral-before-decimal: false
+        forbid-scientific-notation: false
+        forbid-not-a-number: false
+        forbid-infinity: false
+
+.. rubric:: Examples
+
+#. With ``float-values: {require-numeral-before-decimal: true}``
+
+   the following code snippets would **PASS**:
+   ::
+
+    anemometer:
+        angle: 0.0
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+        angle: .0
+
+#. With ``float-values: {forbid-scientific-notation: true}``
+
+   the following code snippets would **PASS**:
+   ::
+
+    anemometer:
+        angle: 0.00001
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+        angle: 10e-6
+
+#. With ``float-values: {forbid-not-a-number: true}``
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+        angle: .NaN
+
+ #. With ``float-values: {forbid-infinity: true}``
+
+   the following code snippets would **FAIL**:
+   ::
+
+    anemometer:
+        angle: .inf
+"""
+
+import re
+
+import yaml
+
+from yamllint.linter import LintProblem
+
+
+ID = "float-values"
+TYPE = "token"
+CONF = {
+    "require-numeral-before-decimal": bool,
+    "forbid-scientific-notation": bool,
+    "forbid-not-a-number": bool,
+    "forbid-infinity": bool,
+}
+DEFAULT = {
+    "require-numeral-before-decimal": False,
+    "forbid-scientific-notation": False,
+    "forbid-not-a-number": False,
+    "forbid-infinity": False,
+}
+
+IS_NUMERAL_BEFORE_DECIMAL_PATTERN = (
+    re.compile("[-+]?(\\.[0-9]+)([eE][-+]?[0-9]+)?")
+)
+IS_SCIENTIFIC_NOTATION_PATTERN = re.compile(
+    "[-+]?(\\.[0-9]+|[0-9]+(\\.[0-9]*)?)([eE][-+]?[0-9]+)"
+)
+IS_INF_PATTERN = re.compile("[-+]?(\\.inf|\\.Inf|\\.INF)")
+IS_NAN_PATTERN = re.compile("\\.nan|\\.NaN|\\.NAN")
+
+
+def check(conf, token, prev, next, nextnext, context):
+    if prev and isinstance(prev, yaml.tokens.TagToken):
+        return
```

Comment:
There's no test for this branch. It should be easy to cover, similarly to what I did in https://github.com/adrienverge/yamllint/pull/461 to cover: https://github.com/adrienverge/yamllint/blob/e49a1011600d8e9831f205e659c389490c2cf7f5/yamllint/rules/octal_values.py#L94-L95

https://github.com/adrienverge/yamllint/blob/82cf1ab222e599e316acb24fefc0ca84ae69229a/tests/rules/test_octal_values.py#L36